Client/Frontend/Browser/TabManager.swift

```diff
@@ -847,8 +847,16 @@ class TabManager: NSObject {
 
   func getTabForURL(_ url: URL) -> Tab? {
     assert(Thread.isMainThread)
-
-    return allTabs.filter { $0.webView?.url == url }.first
+    
+    let tab = allTabs.filter {
+      guard let webViewURL = $0.webView?.url else {
+        return  false
+      }
+      
+      return webViewURL.schemelessAbsoluteDisplayString == url.schemelessAbsoluteDisplayString
```

Comment:
why is this change introduced, any side effects in other places?

Reply:
This change is introduced because the older version (existing) doesnt work and the check for equality always fails.

So we fixed the conditional check here and also I added tests for it,